paimon-core/src/main/java/org/apache/paimon/table/sink/TableCommitImpl.java

```diff
@@ -261,6 +262,9 @@ public void close() throws Exception {
         }
         IOUtils.closeQuietly(lock);
         expireMainExecutor.shutdownNow();
+        if (commit instanceof FileStoreCommitImpl) {
+            ((FileStoreCommitImpl) commit).getCommitMetrics().getMetricGroup().close();
+        }
```

Comment:
Add a `close` method to `FileStoreCommit` interface. Don't forget to check for the usages and close all `commit` object.